src/llmcompressor/utils/metric_logging.py

```diff
@@ -38,6 +44,33 @@ def get_GPU_memory_usage() -> List[Tuple]:
         logger.warning("Failed to obtain GPU usage from pynvml")
         return []
 
+def get_GPU_usage_amd() -> List[Tuple]:
+    usage = []
+    try:
+        import amdsmi
+        try:
+            amdsmi.amdsmi_init()
+            devices = amdsmi.amdsmi_get_processor_handles()
+
+            for device in devices:
+                vram_memory_usage = amdsmi.amdsmi_get_gpu_memory_usage(
+                    device, amdsmi.amdsmi_interface.AmdSmiMemoryType.VRAM
+                )
+                vram_memory_total = amdsmi.amdsmi_get_gpu_memory_total(
+                    device, amdsmi.amdsmi_interface.AmdSmiMemoryType.VRAM
+                )
+
+                memory_percentage = vram_memory_usage / vram_memory_total
+                usage.append(
+                    (memory_percentage, vram_memory_total / (1e9)),
+                )
+            amdsmi.amdsmi_shut_down()
```

Comment:
Have you noticed any slowdowns that might be related to repeatedly calling `amdsmi_init` and `amdsmi_shut_down`?

Reply:
Hmm, No i have not noticed any slowdown as such :)
but yes, the good way is to amdsmi_init, amdsmi_shut_down globally.
I was trying out this, but was a little confused, because these are global functions, hence we might do amdsmi_init() globally, but i am not sure where and when to do amdsmi_shut_down() ?

Reply:
You could check and initialize in `get_GPU_memory_usage()` and just not reinit if it has already been called called. To shutdown, you could have another helper such as `shutdown_gpu_memory_utils()` and have it be called at the end of wherever get_GPU_memory_usage is being called.

This would emulate a contextmanager which we can add in through a followup

Reply:
Theoretically a shutdown call (which cleans up library memory) would not be need, as everything in cleaned up by the python runtime anyways.

As you haven't noticed any slowdowns, let's land as is